tests/metrics/lib/common.bash

```diff
@@ -0,0 +1,354 @@
+		if ! command -v "$cmd" > /dev/null 2>&1; then
+			die "command $cmd not available"
+		fi
+		echo "command: $cmd: yes"
+	done
+}
+
+# This function performs a pull on the image names
+# passed in (notionally as 'about to be used'), to ensure
+#  - that we have the most upto date images
+#  - that any pull/refresh time (for a first pull) does not
+#    happen during the test itself.
+#
+# The image list can be received standalone or as an array, e.g.
+#
+# images=(“img1” “img2”)
+# check_imgs "${images[@]}"
+check_images()
+{
+	local img req_images=( "$@" )
+	for img in "${req_images[@]}"; do
+		echo "ctr pull'ing: $img"
+		if ! sudo "${CTR_EXE}" image pull "$img"; then
+			die "Failed to pull image $img"
+		fi
+		echo "ctr pull'd: $img"
+	done
+}
+
+generate_build_dockerfile() {
+	local dockerfile="$1"
+	local image="$2"
+	local map_key="$3"
+	local text_to_replace="$4"
+	local regs=(${registries["${map_key}"]})
+	for r in ${regs[@]}; do
+		sed 's|'${text_to_replace}'|'${r}'|g' \
+			"${dockerfile}.in" > "${dockerfile}"
+		if sudo "${DOCKER_EXE}" build --build-arg http_proxy="${http_proxy}" --build-arg https_proxy="${https_proxy}" --label "$image" --tag "${image}" -f "$dockerfile" "$dockerfile_dir"; then
+			return 0
+		fi
+	done
+	return 1
+}
+
+# This function performs a build on the image names
+# passed in, to ensure that we have the latest changes from
+# the dockerfiles
+build_dockerfile_image()
+{
+	local image="$1"
+	local dockerfile_path="$2"
+	local dockerfile_dir=${2%/*}
+
+	if [ -f "$dockerfile_path" ]; then
+		echo "docker building $image"
+		if ! sudo "${DOCKER_EXE}" build --build-arg http_proxy="${http_proxy}" --build-arg https_proxy="${https_proxy}" --label "$image" --tag "${image}" -f "$dockerfile_path" "$dockerfile_dir"; then
+			die "Failed to docker build image $image"
+		fi
+		return 0
+	fi
+
+	generate_build_dockerfile "${dockerfile_path}" "${image}" "ubuntu" "@UBUNTU_REGISTRY@" \
+		|| die "Failed to docker build image $image"
+}
+
+# This function removes the ctr image, builds a new one using a dockerfile
+# and imports the image from docker to ctr
+check_ctr_images()
+{
+	local ctr_image="$1"
+	local dockerfile_path="$2"
+	local docker_image="$(echo ${ctr_image} | cut -d/ -f3 | cut -d: -f1)"
+
+	if [ -z "$ctr_image" ] || [ -z "$dockerfile_path" ]; then
+		die "Missing image or dockerfile path variable"
+	fi
+
+	sudo "${CTR_EXE}" i rm "${ctr_image}"
+	build_dockerfile_image "${docker_image}" "${dockerfile_path}"
+	sudo "${DOCKER_EXE}" save -o "${docker_image}.tar" "${docker_image}"
+	sudo "${CTR_EXE}" i import "${docker_image}.tar"
+	rm -rf "${docker_image}".tar
+}
+
+# A one time (per uber test cycle) init that tries to get the
+# system to a 'known state' as much as possible
+metrics_onetime_init()
+{
+	# The onetime init must be called once, and only once
+	if [ ! -z "$onetime_init_done" ]; then
+		die "onetime_init() called more than once"
+	fi
+
+	# Restart services
+	restart_docker_service
```

Comment:
@GabyCT Are we still using docker on metrics?
if not, could you please remove `restart_docker_service`?

Reply:
@dborquez changes applied